api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -80,6 +80,7 @@ type NodePool struct {
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.arch) || has(self.arch)", message="Arch is required once set"
 // +kubebuilder:validation:XValidation:rule="self.arch != 'arm64' || has(self.platform.aws)", message="Setting Arch to arm64 is only supported for AWS"
 // +kubebuilder:validation:XValidation:rule="has(self.replicas) ? !has(self.autoScaling) : has(self.autoScaling)", message="One of replicas or autoScaling should be set but not both"
+// +kubebuilder:validation:XValidation:rule="self.arch != 'ppc64le' || has(self.platform.agent) ||  has(self.platform.powervs)", message="Setting Arch to ppc64le is supported for PowerVSNodePoolPlatform and AgentNodePoolPlatform"
```

Comment:
I'd rather we don't add this CEL, since its only doing part of the validation and the other part is done in the controller with `isArchAndPlatformSupported()`
Splitting validation in 2 places might be confusing and difficult to maintain.

Reply:
Agree, I think keeping this CEL here would intimate the user at the time of creation itself, instead of controller throwing the error at the later point of time.
So IMO, we can keep this CEL and remove the controller validation, Wdyt @muraee @bryan-cox?

Reply:
I'd rather keep `isArchAndPlatformSupported()` since it's more granular on platform/arch choices and its easily unit testable that we have the right combinations.

Reply:
Here are the platform and CPU combos - https://docs.google.com/spreadsheets/d/1sVM28PgAdtmn8AYNihrXuVl7KGcoVxbF-i-i0JU_TPg/edit#gid=1028538637

Reply:
adding CEL validation for the platform/arch combos might get out of hand. 
I am wondering if the `arch` field should just be part of each platformSpec, then we can just restrict the enum directly in the API.

Reply:
That makes sense as an approach though I would see that sort of outside the scope of this PR since we would need to change the HyperShift and HCP CLI.

Reply:
definitely not for this PR.

Reply:
Good idea to keep the arch inside the platform spec and keep the validation there, for now removed the newly added CEL validation and kept only `isArchAndPlatformSupported()` and covered the combinations in sheet too. Ptal now!